lib/smooth/core/network/Wifi.cpp

```diff
@@ -139,10 +141,12 @@ namespace smooth::core::network
             }
             else if (event_id == WIFI_EVENT_AP_START)
             {
+                wifi->ip.addr = 0xC0A80401; // 192.168.4.1
```

Comment:
How about making this a configuration option via menuconfig? Preferably as a string that is parsed if it is possible, or four separate values if need be.

Reply:
I am not sure if it makes sense to make it a configuration option because here the IP address of the interface is not being set; it is set internally in esp-idf to 192.168.4.1 when configured as AP. I did not find any way to read the IP address from esp-idf when the AP mode is started. So, I just set the internal member variable of Wifi to this value (hoping that in future versions of esp-idf this address will not be changed)

Reply:
Ooh, I see. So this is just to keep it in synch with IDF.

Is there really no way to control what IP the AP gets?

Reply:
When using for wifi the [TCP/IP Adapter library](https://docs.espressif.com/projects/esp-idf/en/latest/api-reference/network/tcpip_adapter.html) the IP and MAC can be set using `tcpip_adapter_ap_start.` When using the [Wi-Fi API](https://docs.espressif.com/projects/esp-idf/en/latest/api-reference/network/esp_wifi.html) I did not find a function call to set the IP. 
Since you started the Smooth library using the Wi-Fi API I just continued using it. Maybe it makes sense to switch to the TCP/IP Adapter library later?

Reply:
Yeah, doing that merits its own PR.